airflow/www/views.py

```diff
@@ -839,21 +841,92 @@ def index(self):
         with create_session() as session:
             # read orm_dags from the db
             dags_query = select(DagModel).where(~DagModel.is_subdag, DagModel.is_active)
-
+            dags_query = dags_query.where(DagModel.dag_id.in_(filter_dag_ids))
             if arg_search_query:
-                escaped_arg_search_query = arg_search_query.replace("_", r"\_")
-                dags_query = dags_query.where(
-                    DagModel.dag_id.ilike("%" + escaped_arg_search_query + "%", escape="\\")
-                    | DagModel._dag_display_property_value.ilike(
-                        "%" + escaped_arg_search_query + "%", escape="\\"
+                prefix_search_match = re2.match(r"(?i)(dag|owner|task):\s*", arg_search_query)
+                if prefix_search_match:
+                    query_prefix = prefix_search_match[0].lower()
+                    query_value = arg_search_query[len(query_prefix) :]
+                    if query_value:
+                        escaped_query_value = "%" + query_value.replace("_", r"\_") + "%"
+                        if query_prefix.startswith("task:"):
```

Comment:
Nit: Item 0 is the whole group, item 1 is the first capturing group, and since we have that there's no need to use `.startswith`
```suggestion
                    query_prefix = prefix_search_match[1].lower()
                    query_value = arg_search_query[len(prefix_search_match[0]) :]
                    if query_value:
                        if query_prefix == "task":
```

etc.